docs/_pages/walk-through.md

```diff
@@ -0,0 +1,578 @@
+
+```js
+'use strict';
+
+module.exports = {
+type: 'MainSiteLayout',
+elements() {
+    return [
+    {
+        name: 'headerRow',
+        selector: {
+        type: 'getElementById',
+        value: 'siteHeader',
+        },
+    },
+    {
+    name: 'newsArticle1',
+    selector: {
+        type: 'getElementById',
+        value: 'article1',
+        },
+    },
+    {
+        name: 'newsArticleImage1',
+        selector: {
+        type: 'getElementById',
+        value: 'article1Image',
+        },
+    },
+    {
+        name: 'newsArticleHeading1',
+        selector: {
+        type: 'getElementById',
+        value: 'article1Heading',
+        },
+    },
+    {
+        name: 'newsArticleText1',
+        selector: {
+        type: 'getElementById',
+        value: 'article1Text',
+        },
+    },
+    ];
+},
+```
+
+Above are a list of elements that will be used by the model, we already have our state from our navigation action. Below is the arrangement of those elements into a model, include this in your 'main-site-layout.js':  
+```js
+model() {
+    return {
+        displayed: 'headerRow.isDisplayed',
+        newsArticleImage: {
+        displayed: 'newsArticleImage1.isDisplayed',
+        },
+        newsArticleHeading: {
+            displayed: 'newsArticleHeading1.isDisplayed',
+            text: 'newsArticleHeading1.innerText',
+        },
+        newsArticleText: {
+            displayed: 'newsArticleText1.isDisplayed',
+            text: 'newsArticleText1.innerText',
+        },
+    };
+},
+```
+Finally we have our action and its pieces, include all these in 'main-site-layout.js' as well:
+```js
+    actions() {
+        return {
+            CLICK_TO_VIEW_STORY: {
+```
+Above is the action name, we use app caps with underscores, so just stick with this naming convention, you can name it whatever you want and if you have additional actions, you will name those as additional object properties on the javascript object you are returning as the actions.
+```js
+preconditions() {
+    return [
+        ['isTrue', `pageState.${this.name}.displayed`],
+        ['isTrue', `pageState.${this.name}.newsArticleHeading.displayed`],
+        ['isTrue', `pageState.${this.name}.newsArticleText.displayed`]
+    ];
+},
+```
+The preconditions are what must be true prior to executing the performance of the actions, you do not need to validate everything on the pages as a part of the pre-conditions, but what you might want to validate are properties involved in the performance of the actions or other page state features that might relate directly to your actions. For example, above we are checking that the page is displayed, as well as the news article text and heading, we could also do things like check the image and the text accompanying the article, but that as mentioned above is up to the tester to determine if it is part of what they need to be testing or just extra.
```

Comment:
the half of the last sentence seems a bit wishy washy. Perhaps "but that is left up to the discretion of the tester" would be good?

Reply:
Perhaps use "perform block" or "perform" instead of "performance"